pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -369,6 +369,9 @@ private static PinotQuery compileCalciteSqlToPinotQuery(String sql) {
       DataSource dataSource = new DataSource();
       dataSource.setTableName(fromNode.toString());
       pinotQuery.setDataSource(dataSource);
+      if (fromNode instanceof SqlSelect || fromNode instanceof SqlOrderBy) {
```

Comment:
I do not have deep insight into it. But the following code shows that SqlOrderBy 
 has the SqlSelect member. https://github.com/apache/pinot/blob/master/pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java#L343-L353

Reply:
OK, thanks for the reference. I would suggest changing this logic to:

```
    if (fromNode != null) {
      DataSource dataSource = new DataSource();
      if (fromNode instanceof SqlSelect || fromNode instanceof SqlOrderBy) {
        dataSource.setSubquery(compileSqlNodeToPinotQuery(fromNode));
      } else {
        dataSource.setTableName(fromNode.toString());
      }
      pinotQuery.setDataSource(dataSource);
    }

```

Reply:
Can not leave TableName as null. I will accept your other suggestion.

Reply:
Not sure if I understand. If there is no table name  in the FROM clause, then what is the use of setting the tableName and that too to a string that is actually represents another query? Note that the definition of datasource in query.thrift file allows for tableName being null
```
struct DataSource {
  1: optional string tableName;
  2: optional PinotQuery subquery;
}
```
Also Pinot supports queries without from clause, so even in current code table name (and also subquery) can be null at the same time.

Reply:
Here is the NullPointerException when I made your suggested change.
java.lang.NullPointerException
	at org.apache.pinot.spi.utils.builder.TableNameBuilder.tableHasTypeSuffix(TableNameBuilder.java:73)
	at org.apache.pinot.spi.utils.builder.TableNameBuilder.extractRawTableName(TableNameBuilder.java:100)
	at org.apache.pinot.core.query.reduce.BrokerReduceService.reduceOnDataTable(BrokerReduceService.java:95)
	at org.apache.pinot.queries.BaseQueriesTest.getBrokerResponse(BaseQueriesTest.java:238)
	at org.apache.pinot.queries.BaseQueriesTest.getBrokerResponseForSqlQuery(BaseQueriesTest.java:201)
	at org.apache.pinot.queries.BaseQueriesTest.getBrokerResponseForSqlQuery(BaseQueriesTest.java:173)
	at org.apache.pinot.queries.GapfillQueriesTest.datetimeconvertGapfillTestAggregateAggregateOrderBy(GapfillQueriesTest.java:3554)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:108)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:661)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:869)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1193)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:126)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:744)
	at org.testng.TestRunner.run(TestRunner.java:602)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:380)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:375)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340)
	at org.testng.SuiteRunner.run(SuiteRunner.java:289)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1301)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1226)
	at org.testng.TestNG.runSuites(TestNG.java:1144)
	at org.testng.TestNG.run(TestNG.java:1115)
	at com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)
	at com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:109)

Reply:
My understanding is that by the time reducer code is called, you have flattened all the subqueries into a single query? In that case you would need to appropriately set a table name for reducer (or make other appropriate changes so that the query is semantically correct).

Reply:
No, I did not flatten the subqueries into a single query.